catalog/spatial/geocoding/spatial-geocoding-offline-catalog/src/main/java/org/codice/ddf/spatial/geocoding/index/GeoNamesCatalogIndexer.java

```diff
@@ -238,23 +214,19 @@ private void removeGeoNamesMetacardsFromCatalog(
     if (catalogProviderOptional.isPresent()) {
       CatalogProvider catalogProvider = catalogProviderOptional.get();
 
-      if (StringUtils.isNotBlank(resource)) {
-        filter =
-            filterBuilder.allOf(
-                filterBuilder
-                    .attribute(GeoEntryAttributes.IMPORT_LOCATION)
-                    .is()
-                    .equalTo()
-                    .text(resource),
-                filter);
-      }
+      final Filter removalFilter = geonamesGazetteerFilter();
```

Comment:
It looks like some of the functionality that I added where we would only attempt to delete entries from the specified location has been removed. This was to prevent cases where we receive geonames/gazetteer information from multiple from deleting all of the data vs targeting just a certain source.

Reply:
I changed it because doing a partial delete of the gazetteer index when the `create` argument is `true` violates the `create` argument's contract. If `true`, the index should be fully deleted and recreated with the entries in the supplied resource.

I'll revert this change (it's not related to this ticket anyway) and we can discuss it further.